src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -21,19 +21,49 @@ public Pinecone(String apiKey) {
         this(apiKey, new OkHttpClient());
     }
 
+    public Pinecone(String apiKey, String sourceTag) {
+        this(new PineconeConfig(apiKey), sourceTag, new OkHttpClient());
+    }
+
     public Pinecone(String apiKey, OkHttpClient okHttpClient) {
-        if (apiKey == null || apiKey.isEmpty()) {
-            throw new PineconeValidationException("The API key is required and must not be empty or null");
-        }
-        config = new PineconeConfig(apiKey);
+        this(new PineconeConfig(apiKey), null, okHttpClient);
+    }
+
+    public Pinecone(String apiKey, String sourceTag, OkHttpClient okHttpClient) {
+        this(new PineconeConfig(apiKey), sourceTag, okHttpClient);
+    }
+    public Pinecone(PineconeConfig pineconeConfig) {
+        this(pineconeConfig, null, new OkHttpClient());
+    }
+
+    public Pinecone(PineconeConfig pineconeConfig, String sourceTag) {
+        this(pineconeConfig, sourceTag, new OkHttpClient());
+    }
+
+    public Pinecone(PineconeConfig pineconeConfig, OkHttpClient okHttpClient) {
+        this(pineconeConfig, null, okHttpClient);
+    }
+
+    public Pinecone(PineconeConfig pineconeConfig, String sourceTag, OkHttpClient okHttpClient) {
+        pineconeConfig.setSourceTag(sourceTag);
+        pineconeConfig.validate();
+        config = pineconeConfig;
+
         ApiClient apiClient = new ApiClient(okHttpClient);
-        apiClient.setApiKey(apiKey);
+        apiClient.setApiKey(config.getApiKey());
+        apiClient.setUserAgent(config.getUserAgent());
+
+        if (Boolean.parseBoolean(System.getenv("PINECONE_DEBUG"))) {
+            apiClient.setDebugging(true);
+        }
```

Comment:
We support this same environment variable on the other clients for logging out information about the calls being made. I enabled this here in order to validate the `user-agent` was being sent. I'm fine removing it if we want, but I don't see an issue with leaving it.